[WebGPU] ComputePassEncoder::setBindGroup crashes under some invalid arguments

mwyrzykowski opened the pull request:
#### 5f97395cfff8d3cd7effe4a5a3c31638cc50eccd
<pre>
[WebGPU] ComputePassEncoder::setBindGroup crashes under some invalid arguments
<a href="https://bugs.webkit.org/show_bug.cgi?id=272863">https://bugs.webkit.org/show_bug.cgi?id=272863</a>
&lt;radar://126621325&gt;

Reviewed by Tadeu Zagallo.

Fix validation errors which were largely caught at the Metal framework layer
but would result in release crashes if run with the validation layer enabled.

* LayoutTests/TestExpectations:
* LayoutTests/fast/webgpu/fuzz-272863-expected.txt: Added.
* LayoutTests/fast/webgpu/fuzz-272863.html: Added.
Add regression tests, the test passes if it reaches the end.

* Source/WebGPU/WebGPU/BindGroup.mm:
(WebGPU::Device::createBindGroup):
* Source/WebGPU/WebGPU/CommandEncoder.h:
* Source/WebGPU/WebGPU/CommandEncoder.mm:
(WebGPU::CommandEncoder::ensureBlitCommandEncoder):
(WebGPU::CommandEncoder::beginComputePass):
(WebGPU::CommandEncoder::beginRenderPass):
(WebGPU::CommandEncoder::encoderIsCurrent const):
(WebGPU::CommandEncoder::copyTextureToTexture):
(WebGPU::CommandEncoder::validateFinishError const):
(WebGPU::CommandEncoder::resolveQuerySet):
* Source/WebGPU/WebGPU/ComputePassEncoder.mm:
* Source/WebGPU/WebGPU/Device.mm:
(WebGPU::Device::Device):
* Source/WebGPU/WebGPU/PresentationContextIOSurface.mm:
(WebGPU::PresentationContextIOSurface::configure):
* Source/WebGPU/WebGPU/RenderBundleEncoder.mm:
(WebGPU::makeRenderBundleICBWithResources):
* Source/WebGPU/WebGPU/RenderPassEncoder.mm:

Canonical link: <a href="https://commits.webkit.org/277718@main">https://commits.webkit.org/277718@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ca2fe1e494faf05db950eef5974ff778ec5ec269

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/48278 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/27490 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/51232 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/50966 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/44343 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/33426 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/25010 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/39440 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/48860 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/25185 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/41710 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/20587 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/22665 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/42944 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/6334 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/44623 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/43345 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/52870 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/23325 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/19677 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/46777 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/24590 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/41893 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/45692 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10672 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/25395 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/24313 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

tadeuzagallo commented on LayoutTests/TestExpectations:
```diff
@@ -2090,6 +2090,8 @@ webkit.org/b/139639 [ Debug ] cssom/non-subpixel-scroll-top-left-values.html [ S
 [ Debug ] fast/webgpu/present-without-compute-pipeline.html [ Slow ]
 [ Debug ] fast/webgpu/multidimensional-texture-bounds.html [ Slow ]
+[ Debug ] fast/webgpu/fuzz-272863.html [ Skip ]
+[ Release ] fast/webgpu/fuzz-272863.html [ Pass Failure ]
```
Why is this pass|failure?

mwyrzykowski replied:
Test passes as long as it doesn’t crash. The output can change due to logging from Media frames depending on when they are GC’ed

tadeuzagallo reviewed (approved):
Should we reduce the test case?

mwyrzykowski committed:
[WebGPU] ComputePassEncoder::setBindGroup crashes under some invalid arguments
https://bugs.webkit.org/show_bug.cgi?id=272863
<radar://126621325>

Reviewed by Tadeu Zagallo.

Fix validation errors which were largely caught at the Metal framework layer
but would result in release crashes if run with the validation layer enabled.

* LayoutTests/TestExpectations:
* LayoutTests/fast/webgpu/fuzz-272863-expected.txt: Added.
* LayoutTests/fast/webgpu/fuzz-272863.html: Added.
Add regression tests, the test passes if it reaches the end.

* Source/WebGPU/WebGPU/BindGroup.mm:
(WebGPU::Device::createBindGroup):
* Source/WebGPU/WebGPU/CommandEncoder.h:
* Source/WebGPU/WebGPU/CommandEncoder.mm:
(WebGPU::CommandEncoder::ensureBlitCommandEncoder):
(WebGPU::CommandEncoder::beginComputePass):
(WebGPU::CommandEncoder::beginRenderPass):
(WebGPU::CommandEncoder::encoderIsCurrent const):
(WebGPU::CommandEncoder::copyTextureToTexture):
(WebGPU::CommandEncoder::validateFinishError const):
(WebGPU::CommandEncoder::resolveQuerySet):
* Source/WebGPU/WebGPU/ComputePassEncoder.mm:
* Source/WebGPU/WebGPU/Device.mm:
(WebGPU::Device::Device):
* Source/WebGPU/WebGPU/PresentationContextIOSurface.mm:
(WebGPU::PresentationContextIOSurface::configure):
* Source/WebGPU/WebGPU/RenderBundleEncoder.mm:
(WebGPU::makeRenderBundleICBWithResources):
* Source/WebGPU/WebGPU/RenderPassEncoder.mm:

Canonical link: https://commits.webkit.org/277718@main
